plugin/trino-hive/src/main/java/io/trino/plugin/hive/util/HiveUtil.java

```diff
@@ -1129,6 +1139,23 @@ public static boolean isIcebergTable(Map<String, String> tableParameters)
         return ICEBERG_TABLE_TYPE_VALUE.equalsIgnoreCase(tableParameters.get(ICEBERG_TABLE_TYPE_NAME));
     }
 
+    public static boolean isHudiTable(Table table)
+    {
+        requireNonNull(table, "table is null");
+        Map<String, String> tableParameters = table.getParameters();
+        String inputFormat = table.getStorage().getStorageFormat().getInputFormat();
+
+        if (tableParameters.containsKey(HUDI_TABLE_TYPE_NAME)
+                && (HUDI_TABLE_TYPE_COPY_ON_WRITE.equalsIgnoreCase(tableParameters.get(HUDI_TABLE_TYPE_NAME)) || HUDI_TABLE_TYPE_MERGE_ON_READ.equalsIgnoreCase(tableParameters.get(HUDI_TABLE_TYPE_NAME)))) {
+            return true;
+        }
```

Comment:
I have little concern about this block because the names `type` and values `cow` & `mor` are too general. Can we change to verify the `inputFormat` always?

Reply:
+1 
Input format check is sufficient. However, input format class name could change based on the base file format of Hudi table. For e.g. [HoodieHFileInputFormat](https://github.com/apache/hudi/blob/master/hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/HoodieHFileInputFormat.java). 

If this method is called just once in the lifecycle of a query and if it is ok to incur one filesystem.exists() check then a better way to check for Hudi table is to see if `.hoodie` dir is present under `table.getStorage().getLocation()`. See https://github.com/trinodb/trino/blob/master/plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiMetadata.java#L197

Reply:
Do note that, currently, the check is being made at metastore level and not at the storage level.

Reply:
Hudi type detection on storage level is required `huvi-common` library in context of `trino-hive` plugin, that be overcomplicating just only for redirections of Hudi tables, that will be checked anyway by `trino-hudi` if a query is redirected from hive catalog.
Debug of this place showed that MoR type tables (`_ro` and `_rt` suffixed) don't have type named parameter but have well defined inputFormat (`org.apache.hudi.hadoop.HoodieParquetInputFormat` and `org.apache.hudi.hadoop.realtime.HoodieParquetRealtimeInputFormat` correspondigly). Because of that, table type checks have been removed.